pypesto/visualize/sampling.py

```diff
@@ -91,7 +117,715 @@ def sampling_fval_trace(
+
+    return level_opacities, variable_colors
+
+
+def sampling_prediction_trajectories(
+        ensemble_prediction: EnsemblePrediction,
+        levels: Union[float, Sequence[float]],
+        title: str = None,
+        size: Tuple[float, float] = None,
+        axes: matplotlib.axes.Axes = None,
+        labels: Dict[str, str] = None,
+        axis_label_padding: int = 50,
+        groupby: str = CONDITION,
+        condition_gap: float = 0.01,
+) -> matplotlib.axes.Axes:
+    """Plot MCMC-based prediction confidence intervals for the
+    model states or outputs. One or various confidence levels
+    can be depicted. Plots are grouped by condition.
+
+    Parameters
+    ----------
+    result:
+        The pyPESTO result object with filled sample result.
+    levels:
+        Credibility levels, e.g. [95] for a 95% credibility interval. See the
+        :py:func:`_get_level_percentiles` method for a description of how these
+        levels are handled, and current limitations.
+    title:
+        Axes title.
+    size: ndarray
+        Figure size in inches.
+    axes:
+        Axes object to use.
+    labels:
+        Keys should be ensemble output IDs, values should be the desired
+        label for that output. Defaults to output IDs.
+    axis_label_padding:
+        Pixels between axis labels and plots.
+    groupby:
+        Group plots by `pypesto.predict.constants.OUTPUT` or
+        `pypesto.predict.constants.CONDITION`.
+    condition_gap:
+        Gap between conditions when
+        `groupby == pypesto.predict.constants.CONDITION`.
+
+    Returns
+    -------
+    axes:
+        The plot axes.
+    """
+    if labels is None:
+        labels = {}
+    if len(list(levels)) == 1:
+        levels = list(levels)
+    levels = sorted(levels, reverse=True)
+    # Get the percentiles that correspond to the requested credibility levels.
+    percentiles = [
+        percentile
+        for level in levels
+        for percentile in _get_level_percentiles(level)
+    ]
+
+    summary = ensemble_prediction.compute_summary(percentiles_list=percentiles)
+
+    condition_ids, output_ids = _get_condition_and_output_ids(summary)
+
+    # Set default labels for any unspecified labels.
+    labels = {id_: labels.get(id_, id_) for id_ in condition_ids + output_ids}
+
+    if groupby == CONDITION:
+        n_variables = len(output_ids)
+        variable_names = output_ids
+        n_subplots = len(condition_ids)
+    elif groupby == OUTPUT:
+        n_variables = len(condition_ids)
+        variable_names = condition_ids
+        n_subplots = len(output_ids)
+    else:
+        raise ValueError(f'Unsupported groupby value: {groupby}')
+
+    level_opacities, variable_colors = \
+        _handle_colors(levels=levels, n_variables=n_variables)
+
+    if axes is None:
+        n_row = int(np.round(np.sqrt(n_subplots)))
+        n_col = int(np.ceil(n_subplots / n_row))
+        fig, axes = plt.subplots(n_row, n_col, figsize=size, squeeze=False)
```

Comment:
One could add 
```
        for ax in axes.flat[n_subplots:]:
            ax.remove()
```
to remove empty plots by default. Same for `sampling_parameter_traces()`.